pkg/flavors/postgres/postgres.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"fmt"
+
+	valid "github.com/asaskevich/govalidator"
+	"github.com/tetratelabs/getenvoy/pkg/flavors"
+)
+
+// Define template parameter names
+const endpoint string = "endpoint"
+const inport string = "inport"
+
+// Flavor implements flavor.FlavorConfigTemplate interface
+// and stores config data specific to Postgres template.
+type Flavor struct {
+	// Location of the postgres server
+	endpoint string
+	// Envoy's listener port
+	inport string
+}
+
+var flavor Flavor
+
+func init() {
+	// Set default values.
+	// Default values are not required to be present in cmd line.
+	flavor.inport = "5432"
+	flavors.AddTemplate("postgres", flavor)
+}
+
+// CheckParams verifies that passed template arguments are correct and
+// are sufficient for creating a valid config from template.
+func (Flavor) CheckParams(params map[string]string) (interface{}, error) {
```

Comment:
Based on the contents of this method, may be it would be better to refactor it into `Parse()` function to avoid non-intuitive mutation ?

Reply:
Changed name CheckParseParams and refactored to take receiver and update receiver's state instead of global.